components/omega/src/infra/logging.h

```diff
@@ -0,0 +1,108 @@
+// Omega logging header file.
+
+#ifndef OMEGA_LOG_H
+#define OMEGA_LOG_H
+
+#pragma once
+
+void omega_log_init();
+
+#include <spdlog/sinks/basic_file_sink.h>
+#include <spdlog/spdlog.h>
+
+#include "YAKL.h"
+
+using yakl::Array;
+using yakl::memDevice;
+using yakl::memHost;
+using yakl::styleC;
+using yakl::c::Bounds;
+using yakl::c::parallel_for;
+
+typedef float real;
+typedef Array<real, 1, memDevice, styleC> real1d;
+typedef Array<real, 2, memDevice, styleC> real2d;
+
+#define LOG_LEVEL_TRACE    SPDLOG_LEVEL_TRACE
+#define LOG_LEVEL_DEBUG    SPDLOG_LEVEL_DEBUG
+#define LOG_LEVEL_INFO     SPDLOG_LEVEL_INFO
+#define LOG_LEVEL_WARN     SPDLOG_LEVEL_WARN
+#define LOG_LEVEL_ERROR    SPDLOG_LEVEL_ERROR
+#define LOG_LEVEL_CRITICAL SPDLOG_LEVEL_CRITICAL
+#define LOG_LEVEL_OFF      SPDLOG_LEVEL_OFF
+
+#if !defined(LOG_ACTIVE_LEVEL)
+#define LOG_ACTIVE_LEVEL LOG_LEVEL_INFO
+#endif
+
+#if LOG_ACTIVE_LEVEL <= LOG_LEVEL_TRACE
+#define LOGGER_TRACE(logger, ...) SPDLOG_LOGGER_TRACE(logger, __VA_ARGS__)
```

Comment:
I don't think we need both LOGGER and LOG? Prefer to settle on just the LOG_ interfaces.

Reply:
My intention with LOGGER macros is to allow Omega developer to use their own loggers. spdlog allows to use custom loggers(and sinks). In most of cases, user does not need to know if the LOGGER macros exist. But some Omega subcomponent developers may want to customize their logger to save log(or some data) in a specific format. If this is not the feature that we want to keep in Omega, then I can easily delete the LOGGER macros.

Reply:
Ah, I see. Go ahead and keep the LOGGERs then. Thanks.